### PR TITLE
fix: Include functional tests jar in docker images

### LINKED ISSUE #3989: Docker images do not include testing-tool JARs
**Describe the bug**

When trying to run the testing tool from the docker image of the server or cli (v0.6.0) with:
```
./ksql-test-runner -s ./test.sql -i ./input.json -o ./output.json
```
The response is an error `Could not find or load main class io.confluent.ksql.test.tools.KsqlTestingTool`. This is caused because we do not package the testing tool jars inside the docker image.

**To Reproduce**
Download the 0.6.0 ksqldb-server docker image and try to run the testing tool.

**Expected behavior**
The testing tool should work

**Actual behaviour**
`Could not find or load main class io.confluent.ksql.test.tools.KsqlTestingTool`

### PR DESCRIPTION
**Pull request by vpapavas**
### Description 
Fixes #3989

We need to include the functional tests jar in the docker image. Instead of creating two new modules that will build the two images for server and cli respectively, we decided to build a single image for both server and cli that will be called ksqldb.

This PR adds a new module that is used only for building the ksqldb docker image. (The actual name of the image (ksqldb) will be assigned by the jenkins tool.)

### Testing done 

Tried the docker image and am able to run server, cli and testing tool.

1. Build local image:
```
mvn -Pdocker package -DskipTests -Dspotbugs.skip -Dcheckstyle.skip  -Ddockerfile.skip=false -Dskip.docker.build=false -Ddocker.upstream-tag=master-latest -Ddocker.tag=local.build -Ddocker.upstream-registry='368821881613.dkr.ecr.us-west-2.amazonaws.com/' -pl :ksql-docker -am -T 4
```

2. Try server
```
docker exec -it ksqldb-server bash
> ksql
```

3. Try cli
```
docker exec -it ksqldb-server ksql http://ksqldb-server:8088
```

4. Steps for testing tool (followed with modifications from here: https://docs.confluent.io/current/ksql/docs/developer-guide/ksql-testing-tool.html)

a. Open terminal into container:
```
docker exec -it ksqldb-server bash
```
b. Create files `input.json`, `output.json`, `statement.json` and copy into container
```
docker cp input.json ksqldb-server:.
```

c. Copy test jars into `/usr/share/java/ksql-rest-app/
```
wget https://repo1.maven.org/maven2/junit/junit/4.12/junit-4.12.jar
wget https://repo1.maven.org/maven2/org/hamcrest/hamcrest-all/1.3/hamcrest-all-1.3.jar
```
d. Run testing tool:
```
ksql-test-runner -s statements.sql -i input.json -o output.json
```

e. Output is:
```
	>>>>> Test failed: Invalid key value for topic test_topic.
Expected KeyType: STRING
Actual KeyType: INTEGER, key: 0.
```

Tried same steps with image created from jenkins script.

Tried quickstart (https://ksqldb.io/quickstart.html), currently fails due to : #4400 .

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
**Docs for using the testing tool with docker image need to get created/updated!**
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+version: '2'
```
Comment (agavra) [inline ksql-docker/docker-compose.yml]: What's this dockerfile? Can we keep the old one with all the instructions on how to build and run it locally? Also this one uses the local image, which probably shouldn't be checked in.

File: ksql-docker/pom.xml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    <dependency>
+      <groupId>io.confluent.ksql</groupId>
+      <artifactId>ksql-cli</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>io.confluent.ksql</groupId>
+      <artifactId>ksql-functional-tests</artifactId>
+      <version>${project.version}</version>
```
Comment (agavra) [inline ksql-docker/pom.xml]: do we need to specify the version here?

File: ksql-docker/pom.xml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    <artifactId>ksql-parent</artifactId>
+    <groupId>io.confluent.ksql</groupId>
+    <version>5.5.0-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>ksql-docker</artifactId>
+
+  <properties>
+    <air.main.basedir>${project.parent.basedir}</air.main.basedir>
+    <main-class>io.confluent.ksql.rest.server.KsqlRestApplication</main-class>
```
Comment (agavra) [inline ksql-docker/pom.xml]: 🤔 is this true for both the CLI and the Rest App? Can you test that the image works with our existing docker tutorials, and if not then see what we need to change? I think you asked this question, but is this `main-class` necessary (can we remove it)?

File: ksql-docker/docker-compose.yml (hunk 1 above)
Comment (vpapavas) [reply ksql-docker/docker-compose.yml]: My bad, shouldn't haven been checked in. I thought I had specifically removed it :/

File: ksql-docker/pom.xml (hunk 2 above)
Comment (vpapavas) [reply ksql-docker/pom.xml]: We do, else maven errors out with 
```
[ERROR]   The project io.confluent.ksql:ksql-docker:5.5.0-SNAPSHOT (/Users/vpapavasileiou/Documents/confluent/ksql/ksql-docker/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for io.confluent.ksql:ksql-functional-tests:jar is missing. @ line 50, column 17
```

Comment (agavra) [reply ksql-docker/pom.xml]: Ahhhh.... instead of updating it here, can you update it in the parent POM alongside the other cross-submodule dependencies?

Comment (vpapavas) [reply ksql-docker/pom.xml]: done

File: ksql-docker/pom.xml (hunk 3 above)
Comment (vpapavas) [reply ksql-docker/pom.xml]: Removed the `main-class` and rest. It was needed for enabling executing the created jar via `-jar` which we don't need in the image

**Pull request merged**